### DISCUSSION TITLE
Secret scanning custom pattern generator - beta feedback

### DISCUSSION THREAD

**Discussion by courtneycl** [community/community#75457] [Code Security]
🚢  Secret scanning has a new, AI-powered regular expression generator for custom patterns. Within the existing custom patterns page, GitHub Advanced Security users can launch a generative AI experience where you input a text description of what pattern you would like to detect, include optional example strings that should be detected, and get matching regular expressions in return.

The generator is in a limited beta and access will be granted through a [waitlist](https://github.com/features/preview/security). 

![screenshot of the generator](https://github.com/courtneycl/cocotests-protected/assets/3474250/ccc9d2f6-2e21-49fa-abee-0e30f3771db3)

This discussion is to collect any and all feedback on the results and experience from beta participants. ✨ 

* [Learn more about the generator](https://docs.github.com/en/code-security/secret-scanning/about-the-regular-expression-generator-for-custom-patterns)
* [Sign up for the waitlist](https://github.com/features/preview/security)

**Comment by Volkanugr** [comment]
Hi,

Great for catching missed API keys or test data that shouldn't be there. Software quality is gradually decreasing, similar tools can work in automation.

**Comment by aboodic** [comment]
[](url)

**Comment by NotYevvie** [comment]
Is there a custom generator beta for letting GitHub know what's NOT a secret? A secret scanner change just bricked all of our infra.

https://github.com/orgs/community/discussions/151444

**Comment by joe10832** [comment]
> 🚢 Secret scanning has a new, AI-powered regular expression generator for custom patterns. Within the existing custom patterns page, GitHub Advanced Security users can launch a generative AI experience where you input a text description of what pattern you would like to detect, include optional example strings that should be detected, and get matching regular expressions in return.
> 
> The generator is in a limited beta and access will be granted through a [waitlist](https://github.com/features/preview/security).
> 
> ![screenshot of the generator](https://private-user-images.githubusercontent.com/3474250/281206071-ccc9d2f6-2e21-49fa-abee-0e30f3771db3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA2MjYxNjUsIm5iZiI6MTc2MDYyNTg2NSwicGF0aCI6Ii8zNDc0MjUwLzI4MTIwNjA3MS1jY2M5ZDJmNi0yZTIxLTQ5ZmEtYWJlZS0wZTMwZjM3NzFkYjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMTZUMTQ0NDI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzEwMDFkZDhiMGI5ZWVkMTY3OTI2ODk4OGYyMWNiOGRhZTk2MGIwZWU1MzNmYmNjNWU3NTE2MjVmMDBjYTk1YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fIvpTyFsL-7XPbhdvcTQwLVvFNoSvAhkTPGRWG44Om8)
> 
> This discussion is to collect any and all feedback on the results and experience from beta participants. ✨
> 
> * [Learn more about the generator](https://docs.github.com/en/code-security/secret-scanning/about-the-regular-expression-generator-for-custom-patterns)
> * [Sign up for the waitlist](https://github.com/features/preview/security)
